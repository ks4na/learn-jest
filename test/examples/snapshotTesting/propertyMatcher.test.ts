// test('will fail every time', () => {
//   const user = {
//     createdAt: new Date(),
//     id: Math.floor(Math.random() * 20),
//     name: 'Bob',
//   }

//   expect(user).toMatchSnapshot()
// })

test('use property matchers', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'Bob',
  }

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
    name: 'Bob',
  })
})

export {}
