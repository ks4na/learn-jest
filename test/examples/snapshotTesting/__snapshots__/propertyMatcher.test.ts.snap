// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`use property matchers 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "Bob",
}
`;
